waspc/data/Generator/templates/react-app/src/queries/index.js

```diff
@@ -1,18 +1,16 @@
 import { useQuery as rqUseQuery } from 'react-query'
 
-export const useQuery = (queryFn, queryFnArgs, config) => {
+export function useQuery(queryFn, queryFnArgs, config) {
   if (typeof queryFn !== 'function') {
-    throw new Error('useQuery requires queryFn to be a function.')
+    throw new TypeError('useQuery requires queryFn to be a function.')
   }
   if (!queryFn.queryCacheKey) {
-    throw new Error('queryFn needs to have queryCacheKey property defined.')
+    throw new TypeError('queryFn needs to have queryCacheKey property defined.')
   }
 
-  const rqResult = rqUseQuery({
+  return rqUseQuery({
     queryKey: [queryFn.queryCacheKey, queryFnArgs],
-    queryFn: (_key, args) => queryFn(args),
-    config
+    queryFn: () => queryFn(queryFnArgs),
```

Comment:
Reference: https://react-query.tanstack.com/guides/migrating-to-react-query-3#reactqueryconfigprovider-and-reactquerycacheprovider-have-both-been-replaced-by-queryclientprovider

Reply:
I guess one solution could also be to do
```js
queryFn: ([_key, args]) => queryFn(args),
```
, right?

But instead you decided to just ignore these and provide args directly, since we already know them. I am just wondering if that is ok to do. Why did I not do that before? I think merely to be extra-safe, just in case there is some additional middleware by react-query that does smth. But ok, I think it is almost sure ok to do it like you did.

Reply:
We discussed this in person, _react-query_'s official docs (link above) recommends the closure approach - they don't do anything special with the arguments.